Automattic/woocommerce-payments
# Check that cart total is not zero before rendering express checkout

Fixes #9687 

#### Changes proposed in this Pull Request

 - Validate that cart amount is not zero before rendering Express Checkout Element in the non-block checkout. This prevents Express Checkout errors in case if shopper adds a coupon which grants 100% discount and free shipping.

I've discovered a few other similar code flows, but those appear to be not affected by the case of coupons, as there is nowhere to enter the coupon code, e.g. on the product page.

#### Testing instructions





 - Create a simple product with a value of 20.
 - Create a coupon that will discount 20 and will enable free shipping.
- Inside a GooglePay/ApplePay compatible browser, add the product to cart, and move to checkout.
 - The express checkout button will be rendered and functional.
 - Apply the discount, and that will make the checkout total 0.
 - Express Checkout buttons should disappear after cart total is updated.

<img width="50%" align="center" alt="coupon applied" src="https://github.com/user-attachments/assets/47ce000c-d113-4663-8ded-d3c16d778a84">


 - Remove the coupon.
 - Express Checkout should become available again.

<img width="50%" align="center" alt="coupon removed" src="https://github.com/user-attachments/assets/d70d4111-bca9-452c-82fb-7c99584e5488">

On a separate note, if we reapply coupon, we will get an error notice "Coupon has already been applied!", but the cart update will happen as it should.

-------------------


**Post merge**
